### PR TITLE
change migrate behaviour

### LINKED ISSUE #871: Change migrate behaviour
Change migrate behaviour:
1. move auto migrate to setup
2. fail to run lakeFS if migration no up to date
3. add an option to migrate down to a specific version if does not exist

### PR DESCRIPTION
**Pull request by guy-har**
closes #871

### CODE REVIEW

File: cmd/lakefs/cmd/run.go [hunk 1]
```diff
@@ -53,6 +53,9 @@ var runCmd = &cobra.Command{
 		// validate service names and turn on the right flags
 		dbParams := cfg.GetDatabaseParams()
 
+		if err := db.ValidateSchemaUpToDate(conf.GetDatabaseParams()); err != nil {
+			logger.WithError(err).Error("Failed on schema validation")
```
Comment (nopcoder) [inline cmd/lakefs/cmd/run.go:57]: Need to exit or use Fatal

File: db/connect.go [hunk 2]
```diff
@@ -47,15 +47,6 @@ func ConnectDB(p params.Database) (Database, error) {
 	conn.SetMaxIdleConns(p.MaxIdleConnections)
 	conn.SetConnMaxLifetime(p.ConnectionMaxLifetime)
 
-	if !p.DisableAutoMigrate {
```
Comment (nopcoder) [inline db/connect.go:50]: Missing the code that removes the member from the struct and the loading from the configuration

File: cmd/lakefs/cmd/run.go [hunk 3]
```diff
@@ -53,6 +53,9 @@ var runCmd = &cobra.Command{
 		// validate service names and turn on the right flags
 		dbParams := cfg.GetDatabaseParams()
 
+		if err := db.ValidateSchemaUpToDate(conf.GetDatabaseParams()); err != nil {
```
Comment (nopcoder) [inline cmd/lakefs/cmd/run.go:56]: 1. `conf` and probably it is not you - but all the commands in lakefs should use `cfg`
2. you don't need to call cfg.GetDatabaseParams() you have them in `dbParams` from line 54

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: cmd/lakefs/cmd/run.go (hunk 1 above)
Comment (guy-har) [reply cmd/lakefs/cmd/run.go:57]: Considered doing that,
But then no one could run a previous version even if they wanted.

**Review by nopcoder — APPROVED** [review]
Cool 👍

**Pull request merged**